cow_instrument/DetectorComponent.cpp

```diff
@@ -27,8 +27,9 @@ bool DetectorComponent::equals(const Component &other) const {
 }
 
 void DetectorComponent::registerContents(
-    std::vector<const Detector *> &lookup) const {
-    lookup.push_back(this);
+    std::vector<const Detector *> &lookupDetectors,
+    std::vector<const PathComponent *> &) const {
+  lookupDetectors.push_back(this);
```

Comment:
Without having read further on yet: I assume this ignores the `PathComponent` vector since the detector component is not part of the path.